### PR TITLE
Paper- Maite Arp- c15 

### PR DESCRIPTION
**Pull request by MaiteArp**

### CODE REVIEW

File: swap_meet/clothing.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from swap_meet.item import Item
+
+class Clothing(Item):
+    def __init__(self, condition = None, age = None): # took category out and passed it in through super. Back to my original working way.
```
Comment (CheezItMan) [inline swap_meet/clothing.py:4]: Did you know that you can use [docstrings](https://www.python.org/dev/peps/pep-0257/) to comment a method.  That way it doesn't go too long on one line.

File: swap_meet/decor.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from swap_meet.item import Item
+
+class Decor(Item):
```
Comment (CheezItMan) [inline swap_meet/decor.py:3]: 👍 Looks good

File: swap_meet/electronics.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from swap_meet.item import Item
+
+class Electronics(Item):
```
Comment (CheezItMan) [inline swap_meet/electronics.py:3]: 👍 Looks good

File: swap_meet/item.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        if condition:
+            self.condition = condition
+        self.age = 0
+        if age:
+            self.age = age
+
+    def __str__(self):
+        return "Hello World!"
+
+    def condition_description(self):
```
Comment (CheezItMan) [inline swap_meet/item.py:18]: Do you need the `>=` instead of `==`?

File: swap_meet/item.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+
+class Item:
```
Comment (CheezItMan) [inline swap_meet/item.py:2]: 👍 Looks good

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
```
Comment (CheezItMan) [inline swap_meet/vendor.py:18]:
```suggestion
        return False
```

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in vendor.inventory:
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item) 
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        else:
+            return False
```
Comment (CheezItMan) [inline swap_meet/vendor.py:35]:
```suggestion
        return False
```

File: swap_meet/vendor.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) > 0 and len(vendor.inventory) > 0:
+            first_item = self.inventory[0]
+            friend_item = vendor.inventory[0]
+            return self.swap_items(vendor, first_item, friend_item)
+        else:
+            return False
```
Comment (CheezItMan) [inline swap_meet/vendor.py:43]:
```suggestion
        return False
```

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        best_condition = 0.0
+        best_item = None
+        for item in self.inventory:
+            if item.category == category:
+                if item.condition > best_condition:
+                    best_condition = item.condition
+                    best_item = item
+        return best_item
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:55]: Good reuse of code.

File: tests/test_wave_07_opt.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    item_d = Decor(age=5)
+    item_e = Clothing(age=3)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    newest_item = tai.get_newest()
+
+    assert newest_item.category == "Clothing"
+    assert newest_item.age == pytest.approx(2)
```
Comment (CheezItMan) [inline tests/test_wave_07_opt.py:33]: In general we use `pytest.approx` to check to see if decimal numbers are approximately correct.  Since you're using integers you can use `==`, but it's cool that you're trying these things. 

I would also add a test for:
1.  What happens if a vendor has no items
2. What happens if two items have the same age

File: tests/test_wave_07_opt.py [hunk 11]
```diff
@@ -0,0 +1,63 @@
+import pytest
```
Comment (CheezItMan) [inline tests/test_wave_07_opt.py:1]: I like the tests you've written here.

File: tests/test_wave_07_opt.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    )
+
+    newest_item = tai.get_newest()
+
+    assert newest_item.category == "Clothing"
+    assert newest_item.age == pytest.approx(2)
+
+
+
+def test_swap_by_newest():
```
Comment (CheezItMan) [inline tests/test_wave_07_opt.py:37]: You might also add a test for:

1.  What happens if one or more vendors do not have any items
2. What happens if a vendor has two items of the same age

File: swap_meet/vendor.py [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        my_best = self.get_best_by_category(their_priority)
+        their_best = other.get_best_by_category(my_priority)
+        return self.swap_items(other, my_best, their_best)
+
+########################### This is the optional part #####################################
+
+    def get_newest(self):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:62]: Nice optionals!

**Review by CheezItMan — COMMENTED** [review]
This was an outstanding submission Maite.  You hit all the learning goals and also wrote the optionals and wrote excellent tests for them.  Well done!

I did leave a few comments on style and a few suggestions on your tests to help cover edge-cases.  Let me know if you have questions.
